libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteProgressChangeListener.kt

```diff
@@ -39,8 +39,15 @@ internal class MapRouteProgressChangeListener(
         routeArrow: MapRouteArrow
     ) : this(routeLine, routeArrow, false)
 
-    private var job: Job? = null
+    private val jobControl by lazy { ThreadController.getMainScopeAndRootJob() }
```

Comment:
This job control is canceled only when the `MapboxNavigation` is destroyed. The animator should be canceled when the Map style is changed, ort when the `MapView` is destroyed instead.